Android13 QPR2 Dependency & Misc Updates

theimpulson opened the pull request:
This PR updates dependencies to the latest release (as per AOSP and Gradle) wherever possible.

theimpulson committed:
Bump kotlin to 1.7.20 based on Android 13 QPR2 release

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

theimpulson committed:
Bump to latest stable AGP and Gradle versions

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

theimpulson committed:
Migrate to new plugins and dependency management system

* Ensure every plugin is present in project's plugin block with version
* Switch to full plugin id for jetbrains plugins
* Move plugin and dependency repos to settings

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

theimpulson committed:
Set root project's name to Seedvault

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

theimpulson committed:
Bump dependencies to Android 13 QPR2 release

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

theimpulson committed:
Switch from -Xopt-in to -opt-in

w: Argument -Xopt-in is deprecated. Please use -opt-in instead
w: C:\Users\aayus\StudioProjects\seedvault\storage\demo\src\main\java\de\grobox\storagebackuptester\restore\DemoSnapshotFragment.kt: (34, 31): 'onBackPressed(): Unit' is deprecated. Overrides deprecated member in 'androidx.core.app.ComponentActivity'. Deprecated in Java

w: Argument -Xopt-in is deprecated. Please use -opt-in instead
w: C:\Users\aayus\StudioProjects\seedvault\storage\lib\src\main\java\org\calyxos\backup\storage\backup\BackupService.kt: (35, 9): This is a delicate API and its use requires care. Make sure you fully read and understand documentation of the declaration that is marked as a delicate API.
w: C:\Users\aayus\StudioProjects\seedvault\storage\lib\src\main\java\org\calyxos\backup\storage\restore\RestoreService.kt: (49, 9): This is a delicate API and its use requires care. Make sure you fully read and understand documentation of the declaration that is marked as a delicate API.

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

chirayudesai commented on build.gradle:
```diff
@@ -1,11 +1,11 @@
 buildscript {
     // 1.3.61 Android 11
     // 1.4.30 Android 12
-    // 1.6.10 Android 13
```
Nit: I'd leave this lin in

theimpulson replied:
Done

chirayudesai commented on build.gradle:
```diff
@@ -1,11 +1,11 @@
     // 1.3.61 Android 11
     // 1.4.30 Android 12
-    // 1.6.10 Android 13
+    // 1.7.20 Android 13 (TQ2A)
```
It's usually called QPR2

theimpulson replied:
Thanks, changed

chirayudesai reviewed (dismissed):
We could also rebuild `android.jar` and `libcore.jar`

theimpulson committed:
Bump ktlint Gradle plugin to 11.3.1

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

chirayudesai commented:
I'd like to do a bit of quick testing with this but otherwise it all looks good.

grote commented:
yeah I'd be most worried about the room upgrade to not affect our K/V DB updates.

theimpulson commented:
The libs versions seem to be the same in the latest QPR3

theimpulson commented:
Rebased to fix conflicts

grote commented on storage/lib/Android.bp:
```diff
@@ -40,7 +40,7 @@ android_library {
     },
     kotlincflags: [
-        "-Xopt-in=kotlin.RequiresOptIn",
+        "-opt-in=kotlin.RequiresOptIn",
```
Is this how they do it now? Is that documented somewhere?

theimpulson replied:
This was hinted at in a compile-time warning. As I see it is mentioned here in docs: https://kotlinlang.org/docs/opt-in-requirements.html#module-wide-opt-in

grote commented on gradle/dependencies.gradle:
```diff
@@ -1,7 +1,7 @@
-    ext.room_version = "2.4.0-alpha05"
-    // https://android.googlesource.com/platform/external/protobuf/+/refs/tags/android-13.0.0_r3/java/pom.xml#7
+    // https://android.googlesource.com/platform/prebuilts/sdk/+/refs/tags/android-13.0.0_r32/current/androidx/Android.bp#4586
+    ext.room_version = "2.5.0-alpha01"
```
Too bad they include an early alpha for something so critical.

theimpulson replied:
Does this matter, as it has been used in AOSP for quite a while?

chirayudesai replied:
It's not used extensively in AOSP.

See https://cs.android.com/search?q=androidx.room_room-runtime&start=1

From what we know, Google mostly checks in prebuilts.

@grote Happy to just check in a prebuilt from this in a separate change (and anything else that we don't want to rely on AOSP versions for - if they version is too old, or alpha, or has any other issue).

theimpulson replied:
Removed for unblocking it, added a comment instead

grote replied:
Thanks for removing. I thought about it and actually there's probably no point in removing it. We build this app with AOSP, so whatever is in AOSP gets used. Removing this here just affects our gradle builds which are dev only.

So, the testing would actually need to happen when AOSP bumps its deps. Ideally, we'd also have some integration tests for K/V backups with room. E.g. Some sample data in the repo and we try to read and compare the result with what we expect. Then just running these tests (on-device!) would already give us more peace of mine.

theimpulson replied:
Opened an issue to track things here: https://github.com/seedvault-app/seedvault/issues/548

theimpulson committed:
Drop buildToolsVersion

If you're using Android plugin for Gradle 3.0.0 or higher, your project automatically uses a default version of the build tools that the plugin specifies

Ref: https://developer.android.com/tools/releases/build-tools

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

theimpulson committed:
Sync appropriate protoc artifact for m1

Fixes compilation on M1 macOS. Check https://github.com/grpc/grpc-java/issues/7690 for further info

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

theimpulson committed:
Switch to new Sdk version properties

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

theimpulson committed:
Set appropriate namespace in all modules

Required by new AGP versions

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

theimpulson committed:
Drop packageName from manifests for tests

Tests are Gradle-only and keeping this in manifest results in namespace collision
Ref: https://developer.android.com/build/configure-app-module#change-namespace-for-testing

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

theimpulson committed:
Bump to AGP and Gradle 8.x

Disable new R8 mode and resource namespacing for now to avoid build failures.
These needs to be worked on before they can be enabled.

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

theimpulson commented:
Bumped AGP and Gradle to 8.x, ktlint to latest stable version and other changes to accommodate them

theimpulson committed:
Update possible github actions

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

grote commented:
So this forces me to use JDK 17 and AS Giraffe I guess :/

> The project is using an incompatible version (AGP 8.1.0) of the Android Gradle plugin. Latest supported version is AGP 8.0.2

grote commented:
So after upgrading my local setup, building it and running tests works here. However, there's an issue with the gradle build light theme where the action bar is all black including the text.

theimpulson commented:
> So after upgrading my local setup, building it and running tests works here. However, there's an issue with the gradle build light theme where the action bar is all black including the text.

@grote it seems to be unrelated to this MR. I tried looking and it seems the aforementioned issue is coming from the fact that colors defined for light theme [here](https://github.com/seedvault-app/seedvault/blob/android13/app/src/main/res/values/colors.xml#L7) are using system colors while dark theme colors [here](https://github.com/seedvault-app/seedvault/blob/android13/app/src/main/res/values-night/colors.xml#L7) simply use a properly defined color. I am unsure why it was done that way but maybe we can handle that in a different MR?

theimpulson commented:
> > So after upgrading my local setup, building it and running tests works here. However, there's an issue with the gradle build light theme where the action bar is all black including the text.
> 
> @grote it seems to be unrelated to this MR. I tried looking and it seems the aforementioned issue is coming from the fact that colors defined for light theme [here](https://github.com/seedvault-app/seedvault/blob/android13/app/src/main/res/values/colors.xml#L7) are using system colors while dark theme colors [here](https://github.com/seedvault-app/seedvault/blob/android13/app/src/main/res/values-night/colors.xml#L7) simply use a properly defined color. I am unsure why it was done that way but maybe we can handle that in a different MR?

Issue to keep track: https://github.com/seedvault-app/seedvault/issues/549
